bash_completion

```diff
@@ -2557,6 +2557,15 @@ _complete_as_root()
     [[ $EUID -eq 0 || ${root_command-} ]]
 }
 
+# Complete on available commands, subject to `no_empty_cmd_completion`.
+#
+# @since 2.12
+_comp_compgen_commands()
+{
+    [[ ! ${cur-} ]] && shopt -q no_empty_cmd_completion && return
+    _comp_compgen -- -c
```

Comment:
This does not escape for example spaces in command names in `$PATH` properly. We apparently try that by setting `compopt -o filenames` in some completions (but not all), but that doesn't help.

It was similarly broken before this change too, but I'd like to either fix that while we work on this, or make it consistent if we do/don't try the filenames compopt.

Reply:
I'm not sure how I tested before, but `compopt -o filenames` does seem to help.